tuning/data/data_processors.py

```diff
@@ -383,11 +380,128 @@ def _execute_data_handlers(
               {",".join([e.name for e in DataHandlerType])}'
         )
 
+    def _split_dataset(
+        self,
+        dataset_config: DataSetConfig,
+        loaded_dataset: Union[
+            Dataset, DatasetDict, IterableDataset, IterableDatasetDict
+        ],
+    ) -> Union[DatasetDict, IterableDatasetDict]:
+
+        seed = self.processor_config.seed
+        streaming = self.processor_config.streaming
+
+        train_split = dataset_config.split.get("train")
+        validation_split = dataset_config.split.get("validation")
+
+        if train_split is None or validation_split is None:
+            raise ValueError("Both 'train' and 'validation' splits must be specified!")
+
+        total_split = train_split + validation_split
+        if total_split > 1.0 or total_split == 0:
+            raise ValueError(f"Sum of splits must be in (0,1]. Got {total_split}")
+
+        # Todo : Add support for using multiple existing splits during spliting
+        # Problem : DatasetDict might have multiple splits already, which to use?
+        # One or more than one, or existing splitName other than train
+        # Currently Supports "train" by default
+        use_existing_splits = ["train"]  # For DatasetDict
+
+        if streaming:
+            # For streaming IterableDataset / IterableDatasetDict
+            # Simple stub: either full train or full validation, no partial splits
+            # For example, if train_split >= 1.0, return all as train,
+            # else if validation_split >= 1.0, all as validation
+            # Partial Splits not Supported yet!
+
+            if isinstance(loaded_dataset, IterableDataset):
```

Comment:
Can you remove the heavy code duplication between this and the below function...it seems to be same logic in `DatasetDict` as well.